tests/integration_tests/foreign_key/data/prepare.sql

```diff
@@ -0,0 +1,65 @@
+set @@global.tidb_enable_foreign_key=1;
+set @@foreign_key_checks=1;
+drop database if exists `foreign_key`;
+create database `foreign_key`;
+use `foreign_key`;
+
+-- Check foreign key on delete/update cascade.
+create table t1 (id int key);
+create table t2 (id int key, constraint fk_1 foreign key (id) references t1 (id) on delete cascade on update cascade);
+begin;
+insert into t1 values (1),(2),(3),(4),(5);
+insert into t2 values (1),(2),(3),(4),(5);
+delete from t1 where id in (1,2);
+update t1 set id=id+10 where id =3;
+commit;
+set @@foreign_key_checks=0;
+delete from t1 where id=4;
+set @@foreign_key_checks=1;
+
+-- Check foreign key on delete/update set null.
+create table t3 (id int key);
+create table t4 (id int key, b int, index idx(b), constraint fk_2 foreign key (b) references t3 (id) on delete set null on update set null);
```

Comment:
Could you add some comment about why it has to be `index idx(b)`?

Also it might be an issue to sync_diff in this case.

Reply:
done.